librz/core/disasm.c

```diff
@@ -1131,6 +1142,85 @@ static void ds_build_op_str(RDisasmState *ds, bool print_color) {
 				core->parser->subrel_addr = killme;
 			}
 		}
+
+		// add for str issue
+		if (begin_mark && markinst_num > 1 && strstr(ds->opstr, "0x")) {
+			//calculate: .got beginning addr - next inst addr
+			ut64 got_beginaddr = 0;
+			ut64 rodata_beginaddr = 0;
+			ut64 rodata_size = 0;
+			ut64 next_instaddr = 0;
+
+			RzList *sections;
+			sections = rz_bin_get_sections(core->bin);
+			RzBinSection *s;
+			RzListIter *iter;
+			rz_list_foreach (sections, iter, s) {
+				if (strcmp(s->name, ".got") == 0) {
+					got_beginaddr = s->vaddr;
+				}
+				if (strcmp(s->name, ".rodata") == 0) {
+					rodata_beginaddr = s->vaddr;
+					rodata_size = s->size;
+				}
+			}
+
+			char *numstr = (char *)malloc(sizeof(char) * 11);
+			memset(numstr, 0, sizeof(char) * 11);
+			strncpy(numstr, markinst[1], 10);
+			sscanf(numstr, "%llx", &next_instaddr);
```

Comment:
there is a parsing library for numbers in rizin. check [rz_num_get_input_value](https://github.com/rizinorg/rizin/blob/dev/librz/include/rz_util/rz_num.h#L76) (for example `rz_num_get_input_value(NULL, next_instaddr)`)